### DISCUSSION TITLE
AutoDL support

### DISCUSSION THREAD

**Discussion by vain4us** [jesec/flood#68] [Ideas]
## Summary
Add autodl(-irssi) support for Flood.

## Idea of implementation
Right now, ruTorrent has a plugin for autodl-irssi. [autodl-irssi](https://github.com/autodl-community/autodl-irssi) is able to monitor IRC #announce channels and automatically download torrents that match user defined filters.
Irssi is included in most releases of linux and could be used for these means, or it could use a package like [NPM IRC](https://www.npmjs.com/package/irc). I think adding support for autodl-irssi would be easier than adding a from-scratch autodl plugin/feature. 

Not sure if this feature is going to get loads of support, but it would be nice to have this added.

**Comment by jesec** [comment]
This feature sounds useful. However, while I would gladly and promptly review and merge a potential PR for this feature, I am afraid that I am not able to implement the feature by myself. This is not a workflow that I have used. In fact, I haven’t heard about this till today.

**Comment by vain4us** [comment]
autodl-irssi is a plugin for irssi that is ready to use. The [ruTorrent plugin](https://github.com/autodl-community/autodl-rutorrent) simply sets up filters/trackers/IRC servers through a GUI. autodl-irssi basically does the rest. The Flood feature would feature a menu/GUI that edits the autodl cfg. I don't think you need to mess with irssi a whole lot!

**Comment by jesec** [comment]
Still. When it comes to technical parts of feature implementations, there are much more to consider and nothing is easy. Without experience on the actual workflow and familiarity of the related components, I don’t think I am the best person to do this. The structural and technical difference between ruTorrent and Flood is significant.

**Comment by vain4us** [comment]
> Still. When it comes to technical parts of feature implementations, there are much more to consider and nothing is easy. Without experience on the actual workflow and familiarity of the related components, I don’t think I am the best person to do this. The structural and technical difference between ruTorrent and Flood is significant.

I see, well, I'm not equipped to deal with this either, so I guess we hope for a saviour ;)

**Comment by mirolm** [comment]
There are more specialized softwares for this. I use for example sonarr (it acts like hub and automates other softwares) to follow shows, rtorrent + flood to download them and jackett to grab the torrents. 

I really love the simplicity flood provides - even the RSS feature feels somehow off to me. Still it is upto jesec to decide.

**Comment by vain4us** [comment]
> There are more specialized softwares for this. I use for example sonarr (it acts like hub and automates other softwares) to follow shows, rtorrent + flood to download them and jackett to grab the torrents.
> 
> I really love the simplicity flood provides - even the RSS feature feels somehow off to me. Still it is upto jesec to decide.

I'm familiar with Sonar and Radarr, the only problem with these programs is their speed, they use RSS downloading instead of IRC announce channels, which means they lack speed. Autodl is used for swarm snatching and ratio building, not necessarily for automatically downloading a certain TV show or movie, although it could be used for this as well.

**Reply by tehconz** [reply]
Autodl can be used with sonarr / radarr using the release-push api. [This script is helpful](https://quickbox.io/knowledge-base/sonarr-radarr-autodl-irssi-script/). I set up a very permissive autodl filter that triggers a sonarr script for all TV releases and a radarr script for all movie releases, then sonarr / radarr makes the actual download decision. The extra latency is very small, less than 1 second.

**Comment by jesec** [comment]
Flood does have RSS subscription function. The listening interval can be as small as 1 minute. You can also add automation rules to automatically start to download torrents with different tags and destination assigned. I am using the RSS function a lot recently and I think it is quite useful. It is also nice that the function is well integrated into Flood. 

I checked the autodl project and go through some tracker configurations. I think most trackers provide the same function over feed subscription as well. May I know what's your unique use case that requires IRC?

**Comment by vain4us** [comment]
> Flood does have RSS subscription function. The listening interval can be as small as 1 minute. You can also add automation rules to automatically start to download torrents with different tags and destination assigned. I am using the RSS function a lot recently and I think it is quite useful. It is also nice that the function is well integrated into Flood.
> 
> I checked the autodl project and go through some tracker configurations. I think most trackers provide the same function over feed subscription as well. May I know what's your unique use case that requires IRC?

1 minute is still quite long compared to the IRC times. IRC will be as quickly as under 1 second, which gives you time to compete with the swarm. The RSS feed with a 1 minute interval will not give you this advantage.

**Comment by jesec** [comment]
It would be difficult to justify the increased complexity to gain the one-minute advantage.
